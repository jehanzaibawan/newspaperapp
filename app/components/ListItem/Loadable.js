/**
 *
 * Asynchronously loads the component for ListItem
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
