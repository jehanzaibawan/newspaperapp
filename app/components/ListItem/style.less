.npApp_list-item {
  display: flex;
  width: 100%;
  flex-direction: row;
  margin: 1rem 0;

  .list-item-thumbnail {
    display: flex;
    width: 20%;
    justify-content: center;
  }

  .list-item-details {
    display: flex;
    flex-direction: column;
    width: 65%;

    .title {
      display: flex;
      width: 100%;

      span {
        font-size: 1rem;
        color: #3c3c3c;
      }
    }

    .sub-title {
      display: flex;
      width: 100%;
      margin-top: 0.7rem;

      span {
        font-size: 0.9rem;
        color: #a6a6a6;
      }
    }

    .info {
      display: flex;
      width: 100%;
      flex-direction: row;
      justify-content: space-between;
      margin-top: 0.3rem;

      .author {
        display: flex;
        font-size: 0.8rem;
        color: #a6a6a6;
      }
    }
  }

  .list-item-arrow {
    display: flex;
    width: 15%;
    justify-content: center;
  }
}