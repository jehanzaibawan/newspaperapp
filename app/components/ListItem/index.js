/**
 *
 * ListItem
 *
 */

import React, { memo, PureComponent } from 'react';
import ClampLines from 'react-clamp-lines';
import Thumbnail from 'components/Thumbnail';
import DateWithIcon from 'components/DateWithIcon';
import ClickableArrow from 'components/ClickableArrow';
import './style.less';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class ListItem extends PureComponent {
  static propTypes = {
    thumbnailSrc: PropTypes.string,
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    publishedOn: PropTypes.string.isRequired,
  };

  onClickHandler = () => {};

  render() {
    const { thumbnailSrc, title, subTitle, author, publishedOn } = this.props;
    return (
      <div className="npApp_list-item">
        <div className="list-item-thumbnail">
          <Thumbnail source={thumbnailSrc} />
        </div>
        <div className="list-item-details">
          <ClampLines
            text={title}
            id="title"
            lines={2}
            ellipsis="..."
            className="title"
            innerElement="span"
            buttons={false}
          />
          <ClampLines
            text={subTitle}
            id="sub-title"
            lines={1}
            ellipsis="..."
            className="sub-title"
            innerElement="span"
            buttons={false}
          />
          <div className="info">
            <div className="author">{author}</div>
            <DateWithIcon publishedOn={publishedOn} />
          </div>
        </div>
        <ClickableArrow
          className="list-item-arrow"
          onClick={this.onClickHandler}
        />
      </div>
    );
  }
}

export default memo(ListItem);
