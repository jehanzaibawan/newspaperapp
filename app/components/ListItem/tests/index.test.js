/**
 *
 * Tests for ListItem
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render, fireEvent } from 'react-testing-library';

import ListItem from '../index';

describe('<ListItem />', () => {
  let props;

  beforeEach(() => {
    props = {
      thumbnailSrc: 'fake-image-source',
      title: 'fake-title',
      subTitle: 'fake-sub-title',
      author: 'fake-author-name',
      publishedOn: '2019-01-01',
    };
  });

  describe('<ListItem />', () => {
    it('Should render and match the snapshot', () => {
      const {
        container: { firstChild },
      } = render(<ListItem {...props} />);
      expect(firstChild).toMatchSnapshot();
    });
  });

  describe('onClickHandler()', () => {
    it('assert there are 6 regions', () => {
      const { getByTestId } = render(<ListItem {...props} />);
      fireEvent.click(getByTestId('list-item-arrow'));
      expect(document.querySelectorAll('.list-item-arrow').length).toBe(1);
    });
  });
});
