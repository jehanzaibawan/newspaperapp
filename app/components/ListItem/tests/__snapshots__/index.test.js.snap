// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> <ListItem /> Should render and match the snapshot 1`] = `
<div
  class="npApp_list-item"
>
  <div
    class="list-item-thumbnail"
  >
    <div
      class="npApp_thumbnail"
    >
      fake-image-source
    </div>
  </div>
  <div
    class="list-item-details"
  >
    <div
      class="clamp-lines title"
    >
      <span
        aria-hidden="true"
        id="clamped-content-title"
      >
        fake-title
      </span>
    </div>
    <div
      class="clamp-lines sub-title"
    >
      <span
        aria-hidden="true"
        id="clamped-content-sub-title"
      >
        fake-sub-title
      </span>
    </div>
    <div
      class="info"
    >
      <div
        class="author"
      >
        fake-author-name
      </div>
      <div
        class="npApp_date-with-icon"
      >
        <span />
        2019-01-01
      </div>
    </div>
  </div>
  <div
    class="npApp_clickable-arrow list-item-arrow"
  >
    <span
      data-testid="list-item-arrow"
      role="button"
      tabindex="0"
    />
  </div>
</div>
`;
